src/ucs/sys/ucs_rtnetlink.c

```diff
@@ -0,0 +1,135 @@
+/**
+* Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2024. ALL RIGHTS RESERVED.
+*
+* See file LICENSE for terms.
+*/
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <ucs/debug/log.h>
+#include <ucs/debug/memtrack_int.h>
+#include <ucs/sys/ucs_rtnetlink.h>
+#include <ucs/sys/ucs_netlink.h>
+#include <ucs/sys/sock.h>
+#include <ucs/type/status.h>
+
+#include <errno.h>
+#include <linux/rtnetlink.h>
+
+#define NETLINK_MESSAGE_MAX_SIZE 8195
```

Comment:
can we get it from a system header file? why is it the right value?

Reply:
@yosefe I couldn't find a good enough prediction for the rt-netlink message size nor a definition from a system header file, so I did a couple of things:
  - I used netlink usage examples and saw that 8195 was a pretty common max size
  - I measured routing table (rt-netlink response) max sizes on various machines and it didn't exceed this number. The maximum size I found was half of it.

Reply:
is there way to increase the size at runtime if it is not enough, or handle incoming message in chunks?

Reply:
[### ](https://www.perplexity.ai/search/netlink-message-max-size-8k-bafGz1cIQYCUtsqMhi42dg)

The Kernel safely assumes it's 8K. Do we really need to accommodate for size increase?

Reply:
Thanks @gleon99! I'll add a comment to clarify the size.

@yosefe I had a few other versions of the buffer allocation and size determination and after some discussions (with @shasson5 as well) we decided it's not worth the 2 requests required (in any case, even when the buffer is originally enough because you need to flush the message from the socket) and that this solution will support the majority of cases. So considering @gleon99's comment I guess we can be even more sure about it.

Reply:
according to the comment in https://github.com/torvalds/linux/blob/master/include/linux/netlink.h#L265, seems it is the limit for a single message, not necessarily the entire payload of the reply data, which could span multiple messages.
we should add some protection/error/warning if we're not able to receive the entire reply in 8K buffer. Since this PR is already big enough, we can solve this in next PR.